File: lib/bean/goods_entity.dart
import 'package:flutter_utils/generated/json/base/json_field.dart';
import 'package:flutter_utils/generated/json/goods_entity.g.dart';
import 'dart:convert';
export 'package:flutter_utils/generated/json/goods_entity.g.dart';

@JsonSerializable()
class GoodsEntity {
	bool? isCollection = false;
	String? goodsName = '';

	GoodsEntity();

	factory GoodsEntity.fromJson(Map<String, dynamic> json) => $GoodsEntityFromJson(json);

	Map<String, dynamic> toJson() => $GoodsEntityToJson(this);

	@override
	String toString() {
		return jsonEncode(this);
	}
}

File: android/app/src/main/kotlin/com/wjf/flutter_utils/GoodsEntity.kt
package com.wjf.flutter_utils

/**
 * @Description
 * @Author WuJianFeng
 * @Date 2024/3/12 9:09
 *
 */

data class GoodsEntity(
    var isCollection: Boolean,
    var goodsName: String,
){
    override fun toString(): String {
        return "{\"isCollection\":\"" + isCollection + "\"" +
                ",\"goodsName\":\"" + goodsName + "\"}"
    }
}